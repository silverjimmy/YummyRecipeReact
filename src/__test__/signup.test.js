import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import expect from "expect";
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import SignUp from '../components/signup';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import FlatButton from 'material-ui/FlatButton';
import toJson from 'enzyme-to-json';


configure({ adapter: new Adapter() });

let match = {
    params : '1'
}

const setUp = () => {
    const props = {
        match
    };
    return shallow(<SignUp {...props} />);
};

describe('<SignUp />', () => {
    const event={
        target:{
            value:{}
        },
        preventDefault: () => {

        },
        push: (event) =>{

        }
    }
    const preventDefault = jest.fn();
    const wrapper = shallow( <SignUp /> );
    it('matches  snapshot', () => {
            const register = setUp;
           expect(shallowToJson(register)).toMatchSnapshot();
        });

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <MuiThemeProvider>
        <SignUp />
        </MuiThemeProvider>, div);
    });

    it('renders the correct form fields', () => {
        expect(wrapper.find('username').length).toBe(0);
        expect(wrapper.find('password').length).toBe(0);
        expect(wrapper.find('#login').length).toBe(0);
    });
    
    it('renders a form element', () => {
        expect(wrapper.find('form').length).toBe(1);
    });
    it('should render <div> without throwing an error', () => {
        expect(wrapper.exists(<h4 />))
    });
    it('should render <div> without throwing an error', () => {
        expect(wrapper.exists(<div className="modal-header"/>))
    });
    it('should render <div> without throwing an error', () => {
        expect(wrapper.exists(<div className="error" />))
    });
    it('input should respond to change event and change the state', () => {
        wrapper.find('#password').simulate('change', { target: { name: 'password', value: 'admin' } });
        expect(wrapper.state('password')).toEqual('admin')
    });
    it('renders login form and submits data', () =>{
        wrapper.find("#signup_form").simulate('submit', {preventDefault(){}});
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    it('render the two inputs', () =>{
        expect(wrapper.find("#password")).toHaveLength(1);
        expect(wrapper.find("#username")).toHaveLength(1);
        expect(wrapper.find("#email")).toHaveLength(1);
        wrapper.find("#password").simulate('change', {target: {value: 'admin'}});
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('make sure the email is valid', () =>{
      wrapper.find("#email").simulate('change', {target: {value: 'geofrocker@gmail.com'}});
      expect(toJson(wrapper)).toMatchSnapshot();
  })
  it('fetch list', () => {
    const wrapper = shallow(<SignUp />); 
    wrapper.instance().handleSubmit(event)
});
    it('input should respond to change event and change the state', () => {
        wrapper.find('#password').simulate('change', { target: { name: 'password', value: 'admin' } });
        expect(wrapper.state('password')).toEqual('admin')
    });
    it('renders login form and submits data', () =>{
        wrapper.find("#signup_form").simulate('submit', {preventDefault(){}});
        expect(toJson(wrapper)).toMatchSnapshot();
    });

})
