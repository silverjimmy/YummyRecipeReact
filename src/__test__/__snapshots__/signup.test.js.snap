// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUp /> make sure the email is valid 1`] = `
<div>
  <section
    className="tiles"
  >
    <article
      className="style7"
    >
      <span
        className="image"
      >
        <img
          alt=""
          src="/images/pic13.jpg"
        />
      </span>
      <a>
        <h2>
           Yummy Recipes 
        </h2>
        <div
          className="content"
        >
          <h4>
             Sign Up
          </h4>
          <form
            action=""
            id="signup_form"
            method="POST"
          >
            <div
              className="error"
            >
              <Dialog
                actions={
                  Array [
                    <FlatButton
                      disabled={false}
                      fullWidth={false}
                      label="Cancel"
                      labelPosition="after"
                      labelStyle={Object {}}
                      onKeyboardFocus={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      onTouchStart={[Function]}
                      onTouchTap={[Function]}
                      primary={true}
                      secondary={false}
                  />,
                    <FlatButton
                      disabled={false}
                      fullWidth={false}
                      label="Discard"
                      labelPosition="after"
                      labelStyle={Object {}}
                      onKeyboardFocus={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      onTouchStart={[Function]}
                      onTouchTap={[Function]}
                      primary={true}
                      secondary={false}
                  />,
                  ]
                }
                autoDetectWindowHeight={true}
                autoScrollBodyContent={false}
                modal={false}
                onRequestClose={[Function]}
                open={false}
                repositionOnUpdate={true}
              />
            </div>
            <div>
              <TextField
                disabled={false}
                floatingLabelFixed={false}
                fullWidth={false}
                hintText="username"
                id="username"
                multiLine={false}
                name="username"
                onChange={[Function]}
                rows={1}
                type="text"
                underlineShow={true}
              />
            </div>
            <div>
              <TextField
                disabled={false}
                floatingLabelFixed={false}
                fullWidth={false}
                hintText="Email"
                id="email"
                multiLine={false}
                name="email"
                onChange={[Function]}
                rows={1}
                type="text"
                underlineShow={true}
              />
            </div>
            <div>
              <TextField
                disabled={false}
                floatingLabelFixed={false}
                fullWidth={false}
                hintText="password"
                id="password"
                multiLine={false}
                name="password"
                onChange={[Function]}
                rows={1}
                type="password"
                underlineShow={true}
              />
            </div>
            <FlatButton
              disabled={false}
              fullWidth={false}
              label="Sign Up"
              labelPosition="after"
              labelStyle={Object {}}
              onClick={[Function]}
              onKeyboardFocus={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onTouchStart={[Function]}
              primary={false}
              secondary={false}
              type="submit"
            />
          </form>
        </div>
      </a>
    </article>
  </section>
</div>
`;

exports[`<SignUp /> matches  snapshot 1`] = `null`;

exports[`<SignUp /> render the two inputs 1`] = `
<div>
  <section
    className="tiles"
  >
    <article
      className="style7"
    >
      <span
        className="image"
      >
        <img
          alt=""
          src="/images/pic13.jpg"
        />
      </span>
      <a>
        <h2>
           Yummy Recipes 
        </h2>
        <div
          className="content"
        >
          <h4>
             Sign Up
          </h4>
          <form
            action=""
            id="signup_form"
            method="POST"
          >
            <div
              className="error"
            >
              <Dialog
                actions={
                  Array [
                    <FlatButton
                      disabled={false}
                      fullWidth={false}
                      label="Cancel"
                      labelPosition="after"
                      labelStyle={Object {}}
                      onKeyboardFocus={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      onTouchStart={[Function]}
                      onTouchTap={[Function]}
                      primary={true}
                      secondary={false}
                  />,
                    <FlatButton
                      disabled={false}
                      fullWidth={false}
                      label="Discard"
                      labelPosition="after"
                      labelStyle={Object {}}
                      onKeyboardFocus={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      onTouchStart={[Function]}
                      onTouchTap={[Function]}
                      primary={true}
                      secondary={false}
                  />,
                  ]
                }
                autoDetectWindowHeight={true}
                autoScrollBodyContent={false}
                modal={false}
                onRequestClose={[Function]}
                open={false}
                repositionOnUpdate={true}
              />
            </div>
            <div>
              <TextField
                disabled={false}
                floatingLabelFixed={false}
                fullWidth={false}
                hintText="username"
                id="username"
                multiLine={false}
                name="username"
                onChange={[Function]}
                rows={1}
                type="text"
                underlineShow={true}
              />
            </div>
            <div>
              <TextField
                disabled={false}
                floatingLabelFixed={false}
                fullWidth={false}
                hintText="Email"
                id="email"
                multiLine={false}
                name="email"
                onChange={[Function]}
                rows={1}
                type="text"
                underlineShow={true}
              />
            </div>
            <div>
              <TextField
                disabled={false}
                floatingLabelFixed={false}
                fullWidth={false}
                hintText="password"
                id="password"
                multiLine={false}
                name="password"
                onChange={[Function]}
                rows={1}
                type="password"
                underlineShow={true}
              />
            </div>
            <FlatButton
              disabled={false}
              fullWidth={false}
              label="Sign Up"
              labelPosition="after"
              labelStyle={Object {}}
              onClick={[Function]}
              onKeyboardFocus={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onTouchStart={[Function]}
              primary={false}
              secondary={false}
              type="submit"
            />
          </form>
        </div>
      </a>
    </article>
  </section>
</div>
`;

exports[`<SignUp /> renders login form and submits data 1`] = `
<div>
  <section
    className="tiles"
  >
    <article
      className="style7"
    >
      <span
        className="image"
      >
        <img
          alt=""
          src="/images/pic13.jpg"
        />
      </span>
      <a>
        <h2>
           Yummy Recipes 
        </h2>
        <div
          className="content"
        >
          <h4>
             Sign Up
          </h4>
          <form
            action=""
            id="signup_form"
            method="POST"
          >
            <div
              className="error"
            >
              <Dialog
                actions={
                  Array [
                    <FlatButton
                      disabled={false}
                      fullWidth={false}
                      label="Cancel"
                      labelPosition="after"
                      labelStyle={Object {}}
                      onKeyboardFocus={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      onTouchStart={[Function]}
                      onTouchTap={[Function]}
                      primary={true}
                      secondary={false}
                  />,
                    <FlatButton
                      disabled={false}
                      fullWidth={false}
                      label="Discard"
                      labelPosition="after"
                      labelStyle={Object {}}
                      onKeyboardFocus={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      onTouchStart={[Function]}
                      onTouchTap={[Function]}
                      primary={true}
                      secondary={false}
                  />,
                  ]
                }
                autoDetectWindowHeight={true}
                autoScrollBodyContent={false}
                modal={false}
                onRequestClose={[Function]}
                open={false}
                repositionOnUpdate={true}
              />
            </div>
            <div>
              <TextField
                disabled={false}
                floatingLabelFixed={false}
                fullWidth={false}
                hintText="username"
                id="username"
                multiLine={false}
                name="username"
                onChange={[Function]}
                rows={1}
                type="text"
                underlineShow={true}
              />
            </div>
            <div>
              <TextField
                disabled={false}
                floatingLabelFixed={false}
                fullWidth={false}
                hintText="Email"
                id="email"
                multiLine={false}
                name="email"
                onChange={[Function]}
                rows={1}
                type="text"
                underlineShow={true}
              />
            </div>
            <div>
              <TextField
                disabled={false}
                floatingLabelFixed={false}
                fullWidth={false}
                hintText="password"
                id="password"
                multiLine={false}
                name="password"
                onChange={[Function]}
                rows={1}
                type="password"
                underlineShow={true}
              />
            </div>
            <FlatButton
              disabled={false}
              fullWidth={false}
              label="Sign Up"
              labelPosition="after"
              labelStyle={Object {}}
              onClick={[Function]}
              onKeyboardFocus={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onTouchStart={[Function]}
              primary={false}
              secondary={false}
              type="submit"
            />
          </form>
        </div>
      </a>
    </article>
  </section>
</div>
`;

exports[`<SignUp /> renders login form and submits data 2`] = `
<div>
  <section
    className="tiles"
  >
    <article
      className="style7"
    >
      <span
        className="image"
      >
        <img
          alt=""
          src="/images/pic13.jpg"
        />
      </span>
      <a>
        <h2>
           Yummy Recipes 
        </h2>
        <div
          className="content"
        >
          <h4>
             Sign Up
          </h4>
          <form
            action=""
            id="signup_form"
            method="POST"
          >
            <div
              className="error"
            >
              <Dialog
                actions={
                  Array [
                    <FlatButton
                      disabled={false}
                      fullWidth={false}
                      label="Cancel"
                      labelPosition="after"
                      labelStyle={Object {}}
                      onKeyboardFocus={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      onTouchStart={[Function]}
                      onTouchTap={[Function]}
                      primary={true}
                      secondary={false}
                  />,
                    <FlatButton
                      disabled={false}
                      fullWidth={false}
                      label="Discard"
                      labelPosition="after"
                      labelStyle={Object {}}
                      onKeyboardFocus={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      onTouchStart={[Function]}
                      onTouchTap={[Function]}
                      primary={true}
                      secondary={false}
                  />,
                  ]
                }
                autoDetectWindowHeight={true}
                autoScrollBodyContent={false}
                modal={false}
                onRequestClose={[Function]}
                open={false}
                repositionOnUpdate={true}
              />
            </div>
            <div>
              <TextField
                disabled={false}
                floatingLabelFixed={false}
                fullWidth={false}
                hintText="username"
                id="username"
                multiLine={false}
                name="username"
                onChange={[Function]}
                rows={1}
                type="text"
                underlineShow={true}
              />
            </div>
            <div>
              <TextField
                disabled={false}
                floatingLabelFixed={false}
                fullWidth={false}
                hintText="Email"
                id="email"
                multiLine={false}
                name="email"
                onChange={[Function]}
                rows={1}
                type="text"
                underlineShow={true}
              />
            </div>
            <div>
              <TextField
                disabled={false}
                floatingLabelFixed={false}
                fullWidth={false}
                hintText="password"
                id="password"
                multiLine={false}
                name="password"
                onChange={[Function]}
                rows={1}
                type="password"
                underlineShow={true}
              />
            </div>
            <FlatButton
              disabled={false}
              fullWidth={false}
              label="Sign Up"
              labelPosition="after"
              labelStyle={Object {}}
              onClick={[Function]}
              onKeyboardFocus={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onTouchStart={[Function]}
              primary={false}
              secondary={false}
              type="submit"
            />
          </form>
        </div>
      </a>
    </article>
  </section>
</div>
`;
